enigma-swing/src/main/java/cuchaz/enigma/gui/panels/right/RightRotatedLayerUI.java

```diff
@@ -0,0 +1,362 @@
+package cuchaz.enigma.gui.panels.right;
+
+import javax.swing.JComponent;
+import javax.swing.JLayer;
+import javax.swing.SwingUtilities;
+import javax.swing.plaf.LayerUI;
+import java.awt.AWTEvent;
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.awt.event.MouseMotionListener;
+import java.awt.event.MouseWheelEvent;
+import java.awt.event.MouseWheelListener;
+
+/**
+ * Originally by 2xsaiko.
+ * todo look into replacing this with something simpler
```

Comment:
good luck lmao
(I will actually be impressed if you can find something simpler that isn't extremely janky)

Reply:
glad I have your blessing!
if all else fails I'm just going to hardcore it to 90° rotations, counterclockwise and clockwise.

Reply:
I've been defeated, stackoverflow has nothing for me
your code stays!